service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionController.java

```diff
@@ -49,51 +58,59 @@
     private final CaseValidatorService caseValidatorService;
     private final ObjectMapper mapper;
     private final RestrictionsConfiguration restrictionsConfiguration;
-
-    @Autowired
-    public CaseSubmissionController(
-        UserDetailsService userDetailsService,
-        DocumentGeneratorService documentGeneratorService,
-        UploadDocumentService uploadDocumentService,
-        CaseValidatorService caseValidatorService,
-        ObjectMapper mapper,
-        ApplicationEventPublisher applicationEventPublisher,
-        RestrictionsConfiguration restrictionsConfiguration) {
-        this.userDetailsService = userDetailsService;
-        this.documentGeneratorService = documentGeneratorService;
-        this.uploadDocumentService = uploadDocumentService;
-        this.applicationEventPublisher = applicationEventPublisher;
-        this.caseValidatorService = caseValidatorService;
-        this.mapper = mapper;
-        this.restrictionsConfiguration = restrictionsConfiguration;
-    }
+    private final FeeService feeService;
 
     @PostMapping("/about-to-start")
     public AboutToStartOrSubmitCallbackResponse handleAboutToStartEvent(
         @RequestHeader(value = "authorization") String authorization,
         @RequestBody CallbackRequest callbackRequest) {
+
         CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        Map<String, Object> data = caseDetails.getData();
+        CaseData caseData = mapper.convertValue(data, CaseData.class);
 
-        String label = String.format(CONSENT_TEMPLATE, userDetailsService.getUserName(authorization));
+        List<String> errors = validate(caseData);
+        BigDecimal amount = null;
 
-        Map<String, Object> data = caseDetails.getData();
-        data.put("submissionConsentLabel", label);
+        try {
+            amount = getAmountToPay(caseData.getOrders());
```

Comment:
Shouldnn't we short circuit logic if the validation errors are on case? I don't see why we query Fee register in case of error, we are not adding it to the result anyway. Please consider changing it.